frontend/src/app/pages/exercises/exercise/shared/utility/parse-csv.ts

```diff
@@ -0,0 +1,311 @@
+        const biometricInformation = BiometricInformation.create(
+            splitString[i + 7]!,
+            Number(splitString[i + 5]),
+            splitString[i + 6] === 'M'
+                ? 'male'
+                : splitString[i + 6] === 'W'
+                ? 'female'
+                : 'diverse'
+        );
+
+        // 8, 9 are more organisational data
+        const pretriageInformation = PretriageInformation.create(
+            splitString[i + 28]!,
+            splitString[i + 29]! + splitString[i + 30]!,
+            splitString[i + 56] === 'G',
+            // XABCDE
+            splitString[i + 10]! + splitString[i + 11]!,
+            splitString[i + 12]!,
+            Breathing.create(
+                splitString[i + 13]!,
+                splitString[i + 14]!,
+                splitString[i + 15]!
+            ),
+            Circulation.create(
+                splitString[i + 16]!,
+                splitString[i + 17]!,
+                splitString[i + 18]!,
+                splitString[i + 19]!
+            ),
+            Disability.create(
+                splitString[i + 20]!,
+                splitString[i + 21]!,
+                splitString[i + 22]!,
+                splitString[i + 23]!,
+                splitString[i + 24]!,
+                splitString[i + 27]!
+            ),
+            Exposure.create(splitString[i + 25]!, splitString[i + 26]!)
+        );
+
+        // 31 - 55 are data for images. These are currently ignored
+
+        const statusCode =
+            splitString[i + 61]! +
+            splitString[i + 62]! +
+            splitString[i + 63]! +
+            splitString[i + 64]! +
+            splitString[i + 65]! +
+            splitString[i + 66]!;
+
+        const healthStates = generateHealthStates(
+            splitString.slice(i + 67, i + 104),
+            pretriageInformation
+        );
+
+        const patientTemplate = PatientTemplate.create(
+            biometricInformation,
+            healthStates,
+            defaultImage,
+            'Phase1State'
+        );
+
+        const currentCategory = patientData.find(
+            (data) => data.code === statusCode
+        );
+
+        if (!currentCategory) {
+            patientData.push({
+                code: statusCode,
+                templates: [patientTemplate],
+            });
+        } else {
+            currentCategory.templates.push(patientTemplate);
+        }
+    }
+
+    return patientData;
+}
+
+type ColorCategory = Exclude<PatientStatus, 'white'>;
+class HealthStateData {
+    constructor(
+        public name: string,
+        public status: ColorCategory,
+        public conditions: ConditionParameters[],
+        public startingPhase: number,
+        public finished: boolean
+    ) {}
+}
+
+function generateHealthStates(
+    healthStateInformation: string[],
+    pretriageInformation: PretriageInformation
+) {
+    const phaseTime = 12 * 60 * 1000;
```

Comment:
I have seen this or similar constants (x * 60 *1000)as throughout all the Changes, and I don't really understand them. Maybe I would put them in a combined place.

Reply:
Just because I saw this now: that's 12 minutes (min * s * ms).